bundles/framework/announcements/view/AnnouncementsModal.jsx

```diff
@@ -0,0 +1,53 @@
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Message, Button, Checkbox, Modal } from 'oskari-ui';
+import { Controller, LocaleConsumer } from 'oskari-ui/util';
+import styled from 'styled-components';
+
+//Pop-up functionality for announcements
+
+const StyledCheckbox = styled(Checkbox)`
+    position: absolute !important;
+    left: 0px !important;
+    padding: 4px 15px !important;
+`;
+
+const AnnouncementsModal = ({ id, title, content, controller, index, checked }) => {
+
+    const handleOk = (checked, id) => {
+        controller.setAnnouncementAsSeen(checked, id);
+    }
+
+    const onCheckboxChange = (e) => {
+        controller.onCheckboxChange(e.target.checked);
+    }
+
+    return (
+        <Modal
+              mask={false}
+              centered
+              title={title}
+              visible={true}
+              onCancel={() => handleOk(index)}
+              footer={[
+                <StyledCheckbox key="checkbox" onChange={onCheckboxChange} ><Message messageKey={'dontShow'}/></StyledCheckbox>,
```

Comment:
This onCheckboxChange still feels weird to me since the checked value is not used to toggle the checkbox. But if it works I guess its ok. I have a feeling we might change this at some point anyway so merging for now.